pandas_profiling/model/correlations.py

```diff
@@ -0,0 +1,227 @@
+    """Check if two variables are recoded based on their crosstab.
+
+    Args:
+        confusion_matrix: Crosstab between two variables.
+        count:  The number of variables.
+
+    Returns:
+        Whether the variables are recoded.
+    """
+    return int(confusion_matrix.values.diagonal().sum() == count)
+
+
+def cramers_matrix(df: pd.DataFrame, variables: dict):
+    """Calculate the Cramers v correlation matrix.
+
+    Args:
+        df: The pandas DataFrame.
+        variables: A dict with column names mapped to variable type.
+
+    Returns:
+        A cramers v matrix for categorical variables.
+    """
+    return categorical_matrix(
+        df, variables, partial(cramers_corrected_stat, correction=True)
+    )
+
+
+def recoded_matrix(df: pd.DataFrame, variables: dict):
+    """Calculate the recoded correlation matrix.
+
+    Args:
+        df: The pandas DataFrame.
+        variables: A dict with column names mapped to variable type.
+
+    Returns:
+        A recoded matrix for categorical variables.
+    """
+    return categorical_matrix(df, variables, partial(check_recoded, count=len(df)))
+
+
+def categorical_matrix(
+    df: pd.DataFrame, variables: dict, correlation_function: callable
+):
+    """Calculate a correlation matrix for categorical variables.
+
+    Args:
+        df: The pandas DataFrame.
+        variables: A dict with column names mapped to variable type.
+        correlation_function: A function to calculate the correlation between two variables.
+
+    Returns:
+        A correlation matrix for categorical variables.
+    """
+    categoricals = {
+        column_name: df[column_name]
+        for column_name, variable_type in variables.items()
+        if variable_type == Variable.TYPE_CAT
+        and df[column_name].nunique()
+        <= config["categorical_maximum_correlation_distinct"].get(int)
+    }
+
+    correlation_matrix = pd.DataFrame(
+        np.ones((len(categoricals), len(categoricals))),
+        index=categoricals.keys(),
+        columns=categoricals.keys(),
+    )
+
+    for (name1, data1), (name2, data2) in itertools.combinations(
+        categoricals.items(), 2
+    ):
+        confusion_matrix = pd.crosstab(data1, data2)
+        correlation_matrix.loc[name2, name1] = correlation_matrix.loc[
+            name1, name2
+        ] = correlation_function(confusion_matrix)
+
+    return correlation_matrix
+
+
+def calculate_correlations(df: pd.DataFrame, variables: dict) -> dict:
+    """Calculate the correlation coefficients between variables for the correlation types selected in the config
+    (pearson, spearman, kendall, Phi_k).
+
+    Args:
+        variables: A dict with column names and variable types.
+        df: The DataFrame with variables.
+
+    Returns:
+        A dictionary containing the correlation matrices for each of the active correlation measures.
+    """
+    correlations = {}
+    if config["correlations"]["pearson"].get(bool):
```

Comment:
we can do the first 3 correlation with :
corr_types = [pearson, spearman, kendall]
for corr_type in corr_types:
        correlation = df.corr(method=corr_type)
        if len(correlation) > 0:
            correlations[corr_type] = correlation